File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int solve(int idx , int dp[]){
        if(idx == 0 || idx == 1)return 1;
        if(dp[idx] != -1)return dp[idx];
        int ans = solve(idx-1 , dp) + solve(idx-2 , dp);
        dp[idx] = ans;
        return dp[idx];
    }
    public int climbStairs(int idx) {
        int dp[] =new int[idx+1];
        Arrays.fill(dp ,-1);
       return solve(idx, dp);
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int prev = 1;
        int curr = 1;

        for(int i=2;i<=n;i++){
            int temp = prev + curr;
            prev = curr;
            curr = temp;
        }
        return curr;

    }
};